src/test/java/ezschedule/logic/commands/RecurCommandTest.java

```diff
@@ -0,0 +1,289 @@
+        Date clashDate = new Date(eventToRecur.getDate().date.plusDays(4).toString());
+        String month = validDailyRecurCommandStub.intToStringMonth(clashDate.getMonth());
+
+        // EAT has clashing time with ART
+        Event eat = new EventBuilder().withName("Eating contest")
+                .withDate("2023-05-05").withStartTime("12:00").withEndTime("13:00").build();
+        model.addEvent(eat);
+
+        assertCommandFailure(validDailyRecurCommandStub, model,
+                String.format(RecurCommand.MESSAGE_FAILURE_EVENT_CLASH, clashDate.getDay(), month));
+    }
+
+    @Test
+    public void execute_recurWeeklyWithClash_failure() {
+        Event eventToRecur = model.getFilteredEventList().get(0);
+        Date clashDate = new Date(eventToRecur.getDate().date.plusWeeks(5).toString());
+        String month = validWeeklyRecurCommandStub.intToStringMonth(clashDate.getMonth());
+
+        // eat has clashing time with ART
+        Event eat = new EventBuilder().withName("Eating contest")
+                .withDate("2023-06-05").withStartTime("12:00").withEndTime("13:00").build();
+        model.addEvent(eat);
+
+        assertCommandFailure(validWeeklyRecurCommandStub, model,
+                String.format(RecurCommand.MESSAGE_FAILURE_EVENT_CLASH, clashDate.getDay(), month));
+    }
+
+    @Test
+    public void execute_recurMonthlyWithClash_failure() {
+        Event eventToRecur = model.getFilteredEventList().get(0);
+        Date clashDate = new Date(eventToRecur.getDate().date.plusMonths(5).toString());
+        String month = validMonthlyRecurCommandStub.intToStringMonth(clashDate.getMonth());
+
+        // EAT has clashing time with ART
+        Event eat = new EventBuilder().withName("Eating contest")
+                .withDate("2023-10-01").withStartTime("12:00").withEndTime("13:00").build();
+        model.addEvent(eat);
+
+        assertCommandFailure(validMonthlyRecurCommandStub, model,
+                String.format(RecurCommand.MESSAGE_FAILURE_EVENT_CLASH, clashDate.getDay(), month));
+    }
+
+    @Test
+    public void execute_validIntToStringMonths_success() {
+        // valid months [1, 12]
+        String expectedMonthJan = "January";
+        String actualMonthJan = validDailyRecurCommandStub.intToStringMonth(1);
+
+        String expectedMonthFeb = "February";
+        String actualMonthFeb = validDailyRecurCommandStub.intToStringMonth(2);
+
+        String expectedMonthMar = "March";
+        String actualMonthMar = validDailyRecurCommandStub.intToStringMonth(3);
+
+        String expectedMonthApr = "April";
+        String actualMonthApr = validDailyRecurCommandStub.intToStringMonth(4);
+
+        String expectedMonthJun = "June";
+        String actualMonthJun = validDailyRecurCommandStub.intToStringMonth(6);
+
+        String expectedMonthDec = "December";
+        String actualMonthDec = validDailyRecurCommandStub.intToStringMonth(12);
+
```

Comment:
possible to do for all month? to make sure we get the code coverage.

Reply:
updated for all months

Reply:
There is probably a cleaner way to do this, for example:
```
// pseudo code
arr months = ["offset", "January", "February", ... , "December" ]
for i from 1 to n:
  String expectedMonth = months[i];
  String actualMonth = validDailyRecurCommandStub.intToStringMonth(i);
  assertEquals(expectedMonth, actualMonth);
  
```